plugin-server/tests/worker/ingestion/person-state.test.ts

```diff
@@ -17,14 +23,71 @@ const timestamp = DateTime.fromISO('2020-01-01T12:00:05.200Z').toUTC()
 const timestamp2 = DateTime.fromISO('2020-02-02T12:00:05.200Z').toUTC()
 const timestampch = '2020-01-01 12:00:05.000'
 
+async function fetchPostgresPersonIdOverrides(hub: Hub, teamId: number): Promise<[string, string][]> {
+    const result = await hub.db.postgres.query(
+        PostgresUse.COMMON_WRITE,
+        `
+        WITH overrides AS (
+            SELECT id, old_person_id, override_person_id
+            FROM posthog_personoverride
+            WHERE team_id = ${teamId}
+            ORDER BY id
+        )
+        SELECT
+            mapping.uuid AS old_person_id,
+            overrides_mapping.uuid AS override_person_id
+        FROM
+            overrides AS first
+        JOIN
+            posthog_personoverridemapping AS mapping ON first.old_person_id = mapping.id
+        JOIN (
+            SELECT
+                second.id AS id,
+                uuid
+            FROM
+                overrides AS second
+            JOIN posthog_personoverridemapping AS mapping ON second.override_person_id = mapping.id
+        ) AS overrides_mapping ON overrides_mapping.id = first.id
+        `,
+        undefined,
+        'fetchPersonIdOverrides'
+    )
+    return result.rows.map(({ old_person_id, override_person_id }) => [old_person_id, override_person_id]).sort() as [
+        string,
+        string
+    ][]
+}
+
+interface PersonOverridesMode {
+    getWriter(hub: Hub): PersonOverrideWriter | DeferredPersonOverrideWriter
+    fetchPostgresPersonIdOverrides(hub: Hub, teamId: number): Promise<[string, string][]>
+}
+
+const PersonOverridesModes: Record<string, PersonOverridesMode | undefined> = {
+    disabled: undefined,
+    immediate: {
+        getWriter: (hub) => new PersonOverrideWriter(hub.db.postgres),
+        fetchPostgresPersonIdOverrides: (hub, teamId) => fetchPostgresPersonIdOverrides(hub, teamId),
+    },
+    deferred: {
+        // XXX: This is kind of a mess -- ideally it'd be preferable to just
+        // instantiate the writer once and share it
+        getWriter: (hub) => new DeferredPersonOverrideWriter(hub.db.postgres, 456),
+        fetchPostgresPersonIdOverrides: async (hub, teamId) => {
+            await new DeferredPersonOverrideWriter(hub.db.postgres, 456).processPendingOverrides(hub.db.kafkaProducer)
+            return await fetchPostgresPersonIdOverrides(hub, teamId)
+        },
+    },
+}
```

Comment:
If you just mean the duplication of `DeferredPersonOverrideWriter`, all I can think is that you can make it a class to share that object -- but it doesn't seem like it really matters unless sharing the object is critical because of some state.

Reply:
Yeah, I'd prefer to do something like this but I can't figure out how to correctly type it. 😕 

```diff
diff --git a/plugin-server/tests/worker/ingestion/person-state.test.ts b/plugin-server/tests/worker/ingestion/person-state.test.ts
index 48e35edea0..f4bdeb5232 100644
--- a/plugin-server/tests/worker/ingestion/person-state.test.ts
+++ b/plugin-server/tests/worker/ingestion/person-state.test.ts
@@ -60,24 +60,33 @@ async function fetchPostgresPersonIdOverrides(hub: Hub, teamId: number): Promise
 }
 
 interface PersonOverridesMode {
-    getWriter(hub: Hub): PersonOverrideWriter | DeferredPersonOverrideWriter
-    fetchPostgresPersonIdOverrides(hub: Hub, teamId: number): Promise<[string, string][]>
+    getWriter(): PersonOverrideWriter | DeferredPersonOverrideWriter
+    fetchPostgresPersonIdOverrides(teamId: number): Promise<[string, string][]>
 }
 
-const PersonOverridesModes: Record<string, PersonOverridesMode | undefined> = {
+const PersonOverridesModes: Record<string, ((hub: Hub) => PersonOverridesMode) | undefined> = {
     disabled: undefined,
-    immediate: {
-        getWriter: (hub) => new PersonOverrideWriter(hub.db.postgres),
-        fetchPostgresPersonIdOverrides: (hub, teamId) => fetchPostgresPersonIdOverrides(hub, teamId),
+    immediate: class {
+        constructor(private hub: Hub) {}
+        getWriter() {
+            return new PersonOverrideWriter(this.hub.db.postgres)
+        }
+        fetchPostgresPersonIdOverrides(teamId: number) {
+            return fetchPostgresPersonIdOverrides(this.hub, teamId)
+        }
     },
-    deferred: {
-        // XXX: This is kind of a mess -- ideally it'd be preferable to just
-        // instantiate the writer once and share it
-        getWriter: (hub) => new DeferredPersonOverrideWriter(hub.db.postgres, 456),
-        fetchPostgresPersonIdOverrides: async (hub, teamId) => {
-            await new DeferredPersonOverrideWriter(hub.db.postgres, 456).processPendingOverrides(hub.db.kafkaProducer)
-            return await fetchPostgresPersonIdOverrides(hub, teamId)
-        },
+    deferred: class {
+        private writer: DeferredPersonOverrideWriter
+        constructor(private hub: Hub) {
+            this.writer = new DeferredPersonOverrideWriter(this.hub.db.postgres, 456)
+        }
+        getWriter() {
+            return this.writer
+        }
+        async fetchPostgresPersonIdOverrides(teamId: number) {
+            await this.writer.processPendingOverrides(this.hub.db.kafkaProducer)
+            return await fetchPostgresPersonIdOverrides(this.hub, teamId)
+        }
     },
 }
```

Not a big deal, just kind of clumsy as-is.